.github/actions/docker-build-image/action.yml

```
@@ -0,0 +1,213 @@
+name: Build docker image
+description: Build docker image based on supplied jar files
+inputs:
+  base-directory:
+    description: base working directory directory
+    required: false
+    default: .
+  image-tag:
+    description: tag to be created
+    required: true
+  image-dir:
+    description: Directory holding the Dockerfile. Relative path from base-directory.
+    required: true
+  docker-username:
+    description: Docker.io user name
+    required: true
+  docker-password:
+    description: Docker.io password
+    required: true
+  quay-username:
+    description: Quay.io user name
+    required: true
+  quay-password:
+    description: Quay.io password
+    required: true
+  ghcr-username:
+    description: User name to connect to GHCR
+    required: false
+    default: ${{ github.actor }}
+  ghcr-token:
+    description: Github token to connect to GHCR
+    required: true
+  aws-access-key-id:
+    description: AWS access key id to connect to ECR
+    required: false
+  aws-secret-access-key:
+    description: AWS secret access key to connect to ECR
+    required: false
+  aws-account-id:
+    description: AWS account id to connect to ECR
+    required: true
+  aws-region:
+    description: AWS region to use while pushing to ECR
+    required: false
+    default: us-east-1
+  aws-role-name:
+    required: false
+    description: AWS role name
+    default: ECR_Write
+  grype-scan-enabled:
+    description: Define whether a Grype scan should be executed or not
+    required: false
+    default: 'true'
+  jars-artifact-name:
+    description: Name of the the artifact holding the jar files used to build the docker image
+    required: false
+    default: 'build'
+  preview-label:
+    description: The label name for creating a preview version
+    required: false
+    default: 'preview'
+
+runs:
+  using: composite
+  steps:
+    - name: Set is_preview env variable
+      env:
+        IS_PREVIEW: ${{ github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, inputs.preview-label) }}
+      shell: bash
+      run: |
+        echo "IS_PREVIEW=$IS_PREVIEW" >> $GITHUB_ENV
+
+    - name: set is PUSH_IMAGE env variable
+      env:
+        PUSH_IMAGE: ${{ github.event_name == 'push' || env.IS_PREVIEW == 'true' }}
+      shell: bash
+      run: echo "PUSH_IMAGE=$PUSH_IMAGE" >> $GITHUB_ENV
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: ${{ inputs.jars-artifact-name }}
+
```

```suggestion
```

(this input has been removed from original action recently)

---

Alternative to keep backward compat:

```suggestion
    - uses: actions/download-artifact@v3
      if: inputs.jars-artifact-name != ''
      with:
        name: ${{ inputs.jars-artifact-name }}

```

---

I don't think we need backward compatibility, since we can manage the rollout of the adoption of this action. 
I updated the code with the latest version of `alfresco-process` develop branch and will start updating it. 
If it goes will I'll update the downstream project reference to the action and `workflow/build.yml`